Doc/c-api/intro.rst

```diff
@@ -251,24 +251,31 @@ they are done with the result; this soon becomes second nature.
 Reference Count Details
 ^^^^^^^^^^^^^^^^^^^^^^^
 
-The reference count behavior of functions in the Python/C API is best  explained
+The reference count behavior of functions in the Python/C API is best explained
 in terms of *ownership of references*.  Ownership pertains to references, never
 to objects (objects are not owned: they are always shared).  "Owning a
-reference" means being responsible for calling Py_DECREF on it when the
+reference" means being responsible for calling :c:func:`Py_DECREF` on it when the
 reference is no longer needed.  Ownership can also be transferred, meaning that
 the code that receives ownership of the reference then becomes responsible for
 eventually decref'ing it by calling :c:func:`Py_DECREF` or :c:func:`Py_XDECREF`
 when it's no longer needed---or passing on this responsibility (usually to its
-caller). When a function passes ownership of a reference on to its caller, the
-caller is said to receive a *new* reference.  When no ownership is transferred,
-the caller is said to *borrow* the reference. Nothing needs to be done for a
+caller). Let's look at different types of reference ownership passings in more
+detail.
+
+When a function returns an object and effectively increases the
```

Comment:
I knew that return process itself doesn't affect refcount and that is why I used the word "effectively".
What I meant is this.
"When a function increases the reference count of an object and returns it, the function..."

I thought **a function** can **increase the refcount** of some object in a sense that the reference count can increase during that function calling. If I understood you correctly, I can't say **a function** can **increase or doesn't increase the refcount of some object** because the function is only transferring the ownership of some existing object, right?

How about if I write
"When the reference count of an object is effectively increased during a function call, the function ..."

As a side note, it occurred to me that a function may give ownership of a reference to its caller without having to return that reference in some rare cases. Maybe we don't have to talk about function return at all and the explanation above considers those cases, too.

Reply:
Hummm, I think there may be some confusion because I was not clear in my wording. My comment below applies only to the case when you pass something to a function and then the function can get ownership (increase the refcount) or stealing (don't increase the refcount).

This paragraph that we are commenting here applies to the case when the function returns. In that case, the function can tell you "here is my object, you can use it but is mine and can die unless you take ownership of it (increasing the refcount). But on the other side if you don't need it anymore you don't need to do something because is still mine" or conversely "here is a new object for you, you need to remember to decrease the reference when you no longer need it".

Reply:
Okay, now I see those two cases are different and let's only talk about this paragraph the we are commenting. 

First, please be reminded of the purpose of this PR. If one reads this docs for the first time he has no idea what transferring of ownership actually means and he needs an explanation like this

- Owning a reference: the caller received an object and from the callee which effectively increased the reference count of the object by 1 before returning.
- Borrowing  a reference: the caller received an object and from the callee didn't increase the reference count of the object before returning

Can you see why I keep talking about change of ownership in terms of increasing and decreasing of reference count now?

The only thing I want to make sure here is if it makes sense to say *a function increases or doesn't increase the refcount of the object it returns*.  Can't I say `PyLong_FromLong(1L)` increases the reference count of the object it returns, which is `int(1)`? If not, how do you think I can't explain it without? I can't simply say the caller receives an ownership of  the object  `PyLong_FromLong` returns because it's only a circularly explanation and doesn't help readers.

Reply:
> * Owning a reference: the caller received an object from the callee which had effectively increased the reference count of the object it returned by 1 during function calling

But this is incomplete IMHO, you can receive some object via a borrow reference, increase it and now you own it. Ownership is linked to the duty of decreasing the reference when you are down, not on how did you obtain the object.

Reply:
Well I understand what you mean, but that was not the point of my argument. It seems this conversation is getting more confusing and I think I can just revert this part back to the old version.